### PR TITLE
chore(bundling): use a regular import for `browsers.json`

### PR DESCRIPTION
**Pull request by mhils**
As discussed in #8512: Use a regular `import` for the bundled `browsers.json` so that playwright can be bundled with esbuild et al.

I addressed @aslushnikov's comment. Let me know if you want me to merge the `BrowsersJSON` and `BrowsersJSONDescriptor` typedefs, I left them separate for now as they seem to be slightly different. :)

### CODE REVIEW

**Comment by aslushnikov** [comment]
@mhils something is not happy about your change! https://github.com/microsoft/playwright/pull/8579/checks?check_run_id=3470779520

File: utils/generate_types/test/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "strict": true,
     "target": "ES2019",
     "noEmit": true,
+    "resolveJsonModule": true,
```
Comment (mxschmitt) [inline utils/generate_types/test/tsconfig.json:6]: Can we read it instead of import with require? Thats how we do it for deviceDescriptors

Comment (mhils) [reply utils/generate_types/test/tsconfig.json:6]: I've grepped the repo for `deviceDescriptors`, but only found requires/imports. Could you point me to what you are thinking of?

Comment (mxschmitt) [reply utils/generate_types/test/tsconfig.json:6]: (edited my message, I meant if require works instead of import)

**Comment by aslushnikov** [comment]
Oof, tsconfig changes! I'm worried the scope of the change is way more now. 

@JoelEinbinder: what's your take on adding these flags to the tsconfigs?

**Comment by mhils** [comment]
> Oof, tsconfig changes! I'm worried the scope of the change is way more now.

Yes, agreed. I've taken up @mxschmitt's suggestion from above to use `require`, that avoids the tsconfig changes.

 If you folks still feel this is getting out of hand please do close this PR, I can easily vendor and patch locally. :)

**Review by aslushnikov — APPROVED** [review]

**Comment by aslushnikov** [comment]
Oh this one looks good! LGTM % passed bots

**Pull request merged**